### PR TITLE
Lazy broadcasting for Frequencies

### PR DESCRIPTION
**Pull request by jishnub**
On master:

```julia
julia> fftfreq(4,1) * 2
4-element Array{Float64,1}:
  0.0
  0.5
 -1.0
 -0.5
```

After this PR:

```julia
julia> fftfreq(4, 1) * 2
4-element Frequencies{Float64}:
  0.0
  0.5
 -1.0
 -0.5
```

### CODE REVIEW

File: src/definitions.jl [hunk 1]
```diff
@@ -415,6 +415,11 @@ Base.step(x::Frequencies) = x.multiplier
 
 Base.copy(x::Frequencies) = x
 
+# Retain the lazy representation upon scalar multiplication
+Broadcast.broadcasted(::typeof(*), f::Frequencies, x::Number) = Frequencies(f.n_nonnegative, f.n, f.multiplier * x)
+Broadcast.broadcasted(::typeof(*), x::Number, f::Frequencies) = Broadcast.broadcasted(*, f, x)
+Broadcast.broadcasted(::typeof(/), f::Frequencies, x::Number) = Frequencies(f.n_nonnegative, f.n, f.multiplier / x)
```
Comment (stevengj) [inline src/definitions.jl:421]: I guess for completeness we should also have
```jl
Broadcast.broadcasted(::typeof(\), x::Number, f::Frequencies) = Broadcast.broadcasted(/, f, x)
```

Comment (jishnub) [reply src/definitions.jl:421]: I've added this now

**Comment by stevengj** [comment]
Thanks!

**Comment by jishnub** [comment]
Could you tag a new release?

**Pull request merged**